### PR TITLE
feat: Support latest versions of the plus plugins.

### PR DESCRIPTION
**Pull request by maxlapides**
**Requirements**

- [x] I have added test coverage for new or changed functionality
- [x] I have followed the repository's [pull request submission guidelines](../blob/main/CONTRIBUTING.md#submitting-pull-requests)
- [x] I have validated my changes against all supported platform versions

**Related issues**

It's not currently possible to upgrade the Flutter plus plugins in projects that require the LaunchDarkly SDK until the LaunchDarkly SDK upgrades its dependencies.

**Describe the solution you've provided**

Upgraded the dependencies.

**Describe alternatives you've considered**

N/A

**Additional context**

N/A

### CODE REVIEW

File: packages/flutter_client_sdk/pubspec.yaml [hunk 1]
```diff
@@ -12,11 +12,11 @@ environment:
 dependencies:
   flutter:
     sdk: flutter
-  package_info_plus: ">=4.2.0 <6.0.0"
-  device_info_plus: ^9.1.1
```
Comment (kinyoklion) [inline packages/flutter_client_sdk/pubspec.yaml:16]: It looks like only connectivity plus introduced changes? So this one could be ">=9.1.1 <11.0.0"?

File: packages/flutter_client_sdk/lib/src/flutter_state_detector.dart [hunk 2]
```diff
@@ -23,7 +23,8 @@ final class FlutterStateDetector implements StateDetector {
   Stream<NetworkState> get networkState => _networkStateController.stream;
 
   late final LDAppLifecycleListener _lifecycleListener;
-  late final StreamSubscription<ConnectivityResult> _connectivitySubscription;
```
Comment (kinyoklion) [inline packages/flutter_client_sdk/lib/src/flutter_state_detector.dart:26]: I am a little concerned about also requiring an upgrade to a package that has only been out a few days. 

Ideally we could support both for a bit. Probably not elegant, but I wonder if we could wrap these couple contact points. Accept a dynamic, if the item is not a list, then turn it into a list of one, if it is a list then forward it along.

**Comment by kinyoklion** [comment]
@maxlapides Thank you for the contribution! I've added a couple comments, and we will need a little time to consider.

**Comment by kinyoklion** [comment]
I changed the PR title to ensure that it is releasable via release-please. It is a "feat" which would correspond to a minor version bump.

File: packages/flutter_client_sdk/pubspec.yaml (hunk 1 above)
Comment (maxlapides) [reply packages/flutter_client_sdk/pubspec.yaml:16]: Yes, good idea!

File: packages/flutter_client_sdk/lib/src/flutter_state_detector.dart (hunk 2 above)
Comment (maxlapides) [reply packages/flutter_client_sdk/lib/src/flutter_state_detector.dart:26]: Definitely not elegant, but it's not a bad idea! I just updated this PR, it should do the trick.

I thought about writing a wrapper as you suggested, but it looks like it's really just this one little spot in the code that needs to handle connectivity_plus v5, so I thought I'd keep it simple.

Comment (kinyoklion) [reply packages/flutter_client_sdk/lib/src/flutter_state_detector.dart:26]: @maxlapides Looks great. Thank you for the conrtibution!

**Review by kinyoklion — APPROVED** [review]

**Comment by maxlapides** [comment]
Thanks so much @kinyoklion!

**Pull request merged**